libmobilecoin/libmobilecoin_cbindgen.h

```diff
@@ -381,6 +385,100 @@ ssize_t mc_bip39_entropy_from_mnemonic(FfiStr mnemonic,
  */
 FfiOptOwnedStr mc_bip39_words_by_prefix(FfiStr prefix);
 
+/**
+ * Returns a new ChaCha20Rng instance initialized with the
+ * seed value provided by the u64 long_val parameter
+ *
+ * # Arguments
+ *
+ * * `long_val` - an unsigned 64 bit value to use as the rng seed
+ *
+ * # Errors
+ *
+ * * `LibMcError::Poison`
+ */
+FfiOptOwnedPtr<Mutex<McChaCha20Rng>> mc_chacha20_rng_create_with_long(uint64_t long_val,
```

Comment:
I notice none of the other bindings use `Mutex`, including ones that are definitely not thread-safe. Perhaps the intent is that the thread synchronizations happen on the Java/caller side?

Reply:
In our case, we definitely want this to be thread safe... Not sure of the other C APIs that expose rust instances - it's possible they are all explicitly not threadsafe b/c all the processing that uses the rust instances are within a single thread, or possibly they SHOULD be threadsafe w/a mutex...  In this case, I know we need to be thread-safe.

Resolving this after our Slack chat :)